### PR TITLE
feat: analyze profits - closes #19

### LINKED ISSUE #19: Analyze profits

### PR DESCRIPTION
**Pull request by IlluvatarEru**
Signed-off-by: Arthurim <arthurbdauphine@gmail.com>

## What does this PR do?

A short description of what the PR does.

## Related issue

Link to the issue this PR addresses.

If there isn't already an open issue, create an issue first. This will be our home for discussing the problem itself.

## Testing

What testing was performed to verify this works? Unit tests are a big plus!

## Checklist before merging
- [x] Read the [contributing guide](https://github.com/flashbots/mev-inspect-py/blob/main/CONTRIBUTING.md)
- [x] Installed and ran pre-commit hooks
- [x] All tests pass with `./mev test`

### CODE REVIEW

File: scripts/launch_analysis.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+echo "${reps}"
+for i in $(seq 0 1 $reps)
+do
+    from=$(($blockFrom + $i*$window))
+    to=$(($blockFrom + ($i+1)*$window))
+    echo "--"
+    echo "rep=$i/$reps"
+    echo "from=$from"
+    echo "to=  $to"
+    #./mev inspect-many $from $to
```
Comment (0xpanoramix) [inline scripts/launch_analysis.sh:16]: Can we remove this comment ?

File: profit_analysis/read.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+def read_profit_from_to(db_session, block_from, block_to):
+    where_clause = (
+        "block_number>=" + str(block_from) + " AND " + "block_number<=" + str(block_to)
+    )
+    profit = read_from_db_all_into_dataframe(
+        db_session, "total_profit_by_block", "*", where_clause
+    )
+    profit = pd.DataFrame(profit.fetchall())
+    return profit
```
Comment (0xpanoramix) [inline profit_analysis/read.py:28]: I would move this in `mev-inspect/crud/total_profits.py` as it is a READ operation on the database.

File: profit_analysis/analysis.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        .reset_index()
+    )
+    profit_by_block.columns = profit_by_block.columns.droplevel(0)
+    profit_by_block.rename(columns={"": col}, inplace=True)
+    profit_by_block.to_csv(DATA_PATH + "profit_by_" + col + ".csv", index=False)
+    return profit_by_block
+
+
+def create_web3():
+    w3_provider = web3.Web3(web3.Web3.HTTPProvider("https://polygon-rpc.com"))
```
Comment (0xpanoramix) [inline profit_analysis/analysis.py:217]: How about getting the RPC url as an env. variable ? We never know what might happen.

File: profit_analysis/analysis.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+def analyze_profit(inspect_db_session, block_from, block_to, save_to_csv=False):
+    profit = read_profit_from_to(inspect_db_session, block_from, block_to)
+    w3 = create_web3()
+    profit = add_block_timestamp(w3, profit)
+    profit = add_cg_ids(profit)
+    profit = get_usd_profit_arbitrages(profit, save_to_csv)
+    return profit
+
+
+def get_usd_profit_arbitrages(profit_by_block, save_to_csv=False):
```
Comment (0xpanoramix) [inline profit_analysis/analysis.py:46]: Is it possible to add a quick description of the method ? From a high-level perspective, no need to add lots of detail.
As in: "First, we ... Then, using the results from ... we ... and finally we can ..."
No need to explain edge cases (error management is explicit).

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]
Amazing job ! Just left few minor comments.
Feel free to merge once fixed 🚀

File: profit_analysis/read.py (hunk 2 above)
Comment (IlluvatarEru) [reply profit_analysis/read.py]: :heavy_check_mark:

File: scripts/launch_analysis.sh (hunk 1 above)
Comment (IlluvatarEru) [reply scripts/launch_analysis.sh]: :heavy_check_mark:

File: profit_analysis/analysis.py (hunk 4 above)
Comment (IlluvatarEru) [reply profit_analysis/analysis.py]: :heavy_check_mark:

File: profit_analysis/analysis.py (hunk 3 above)
Comment (IlluvatarEru) [reply profit_analysis/analysis.py]: :heavy_check_mark:

**Review by 0xpanoramix — APPROVED** [review]

**Pull request merged**